test/algorithm/group.hpp

```diff
@@ -0,0 +1,26 @@
+//          Copyright Odin Holmes 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE.md or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+#pragma once
+
+#include <kvasir/mpl/algorithm/group.hpp>
+#include <kvasir/mpl/algorithm/make_sequence.hpp>
+
+namespace {
+	struct group_test {
+		group_test() {
+			using namespace kvasir::mpl;
+
+			eager::group<list<>>{}    = list<>{};
```

Comment:
I like the testing style, I just don't know how well it works in practice since we don't use it anywhere else

Reply:
The goal is to have in the compiler errors the calculated type and the expected type if there is a discrepancy.
I find this much more explicit than a static_assert that doesn't provide this information.

Reply:
the testing style is not bad I think, constructors of empty pods are constexpr as well as copy assignment

Reply:
Alright, then we should probably convert the rest of the testing code to use this style, as it seems better than using static_assert.
The only thing is that the function should still be marked constexpr.

Reply:
agreed

Reply:
constexpr does not work with g++-4.8 and g++-4.9

    The compiler feature "cxx_relaxed_constexpr" is not known to CXX compiler

Add cxx_relaxed_constexpr active also c++14.

Reply:
good point, we really wanted to be c++11 compatible, maybe just lest the tests have runtime component

Reply:
I see what you're getting at. We'll have to decide if it is worth porting the entire codebase to this style, but it seems fine for now.
Just fix the other points I mentioned in my last comment and I'll merge this once the tests pass.